src/virtual_dom/mod.rs

```diff
@@ -39,8 +39,8 @@ type Listeners<COMP> = Vec<Box<dyn Listener<COMP>>>;
 /// A map of attributes.
 type Attributes = HashMap<String, String>;
 
-/// A set of classes.
-type Classes = HashSet<String>;
+/// A list of classes.
+type Classes = Vec<String>;
```

Comment:
We should use an ordered set for fast difference calculation. Could you use BTreeSet instead?